app/controllers/stories_controller.rb

```diff
@@ -36,13 +36,17 @@ def update
     authorize @story
     @story.acting_user = current_user
     @story.base_uri = project_url(@story.project)
+    result = StoryOperations::Update.new.call(story: @story, data: allowed_params, current_user: current_user)
     respond_to do |format|
-      if StoryOperations::Update.call(@story, allowed_params, current_user)
-        format.html { redirect_to project_url(@project) }
-        format.js   { render json: @story }
-      else
-        format.html { render action: 'edit' }
-        format.js   { render json: @story, status: :unprocessable_entity }
+      Dry::Matcher::ResultMatcher.call(result) do |on|
```

Comment:
What do you think about creating an abstraction for that from this point on? Something like:

```rb
  result = StoryOperations::Update.new.call(story: @story, data: allowed_params, current_user: current_user)

  # the abstraction I'm talking about is the method `match_result`
  match_result(result) do |on|
    on.success do |story|
    # ...
  end
```

This can be done on `ApplicationController` (which every controller inherits from)

Reply:
the goal of the abstraction should be just """hide""" the `Dry::Matcher::ResultMatcher`call?

Reply:
Yes

Reply:
done